pandas/core/reshape/pivot.py

```diff
@@ -2,95 +2,28 @@
 
 
 from pandas.core.dtypes.common import is_list_like, is_scalar
+from pandas.core.dtypes.generic import ABCDataFrame, ABCSeries
+
 from pandas.core.reshape.concat import concat
-from pandas import Series, DataFrame, MultiIndex, Index
+from pandas.core.frame import _shared_docs
+# Note: We need to make sure `frame` is imported before `pivot`, otherwise
```

Comment:
hmm I don't like this here the import order shouldn't matter (I know it does, but it shouldn't).

generally we simply import the _shared_docs at the end.